### PR TITLE
Backup/Restore investigation

### PR DESCRIPTION
**Pull request by ryanemerson**

### CODE REVIEW

File: pkg/controller/zerocapacity/zero_controller.go [hunk 1]
```diff
@@ -293,12 +293,10 @@ func (z *Controller) cleanupResources(httpClient http.HttpClient, request reconc
 	var allErrors error
 
 	// Stop the zero-capacity server so that it leaves the Infinispan cluster
-	rsp, err, reason := httpClient.Post(request.Name, consts.ServerHTTPServerStop, "", nil)
+	rsp, err, reason := httpClient.Post(meta.Name, consts.ServerHTTPServerStop, "", nil)
```
Comment (dmvolod) [inline pkg/controller/zerocapacity/zero_controller.go:296]: @ryanemerson it looks like we have duplicate methods for shutdown server or it's temporary solution just for tests as we have `cluster.GracefulShutdown` method

Comment (ryanemerson) [reply pkg/controller/zerocapacity/zero_controller.go:300]: Graceful shuts down the cluster not the individual server.

Comment (dmvolod) [reply pkg/controller/zerocapacity/zero_controller.go:300]: Oh, true, missed this.
In this case we should review some cluster vs individual server interaction and try to optimize it. In some cases Pod[0] can be very heated, but we can interact with cluster via service, as it's much more faster than call curl inside Pod.

Comment (ryanemerson) [reply pkg/controller/zerocapacity/zero_controller.go:300]: In this case we need to speak to a specific pod, the zero-capacity one, so I'm not sure we can go via the service. I could make a Server interface similar to `ClusterInterface` to abstract the details of the REST call for future reuse

Comment (dmvolod) [reply pkg/controller/zerocapacity/zero_controller.go:300]: Yes, sure, this not for current case, just to discuss and remember it.

File: test/e2e/utils/httpClient.go [hunk 2]
```diff
@@ -85,16 +82,13 @@ func (c *httpClientConfig) Post(path, payload string, headers map[string]string)
...
 func (c *httpClientConfig) exec(req *http.Request, headers map[string]string) (*http.Response, error) {
 	if c.isAuthRequired() {
-		if c.authRealm == nil {
-			rsp, err := c.Do(req)
-			ExpectNoError(err)
-			if rsp.StatusCode != http.StatusUnauthorized {
-				return rsp, fmt.Errorf("Expected 401 DIGEST response before content: %v", rsp)
-			}
-			c.authRealm = getAuthorization(*c.username, *c.password, rsp)
+		rsp, err := c.Do(req)
```
Comment (dmvolod) [inline test/e2e/utils/httpClient.go:85]: @ryanemerson probably this situation could happens when later `rsp.Body.Close()` is not calling

File: .travis.yml [hunk 3]
```diff
@@ -31,27 +31,6 @@ jobs:
         - make unit-test
         - make lint
```
Comment (dmvolod) [inline .travis.yml:32]:
```suggestion
```

File: test/e2e/utils/httpClient.go (hunk 2 above)
Comment (dmvolod) [inline test/e2e/utils/httpClient.go:85]: @ryanemerson probably we make `rsp, err := c.Do(req)` calls twice and didn't close first `rsp` and this is the reason why after CR deletion connection still open and tries to invoke on the finish

Comment (ryanemerson) [reply test/e2e/utils/httpClient.go:85]: Good point, I had missunderstood what you meant about the closing of requests previously. Updated

Comment (ryanemerson) [reply test/e2e/utils/httpClient.go]: I think I finally have it :sweat_smile: 

I have added a DEBUG constant to `httpClient.go` that dumps all http req/rsp headers.

**Comment by ryanemerson** [comment]
The backup/restore tests will pass once the `12.1.2.Final` image has been released. As these changes only impact backup/restore tests and these are already failing, this PR should be ok to merge now.

**Review by dmvolod — APPROVED** [review]

**Comment by rigazilla** [comment]
merged thanks @ryanemerson !

**Pull request merged**